runtime/lua/vim/snippet.lua

```diff
@@ -477,7 +477,9 @@ end
 function M.expand(input)
   local snippet = G.parse(input)
   local snippet_text = {}
-  local base_indent = vim.api.nvim_get_current_line():match('^%s*') or ''
+  local curbuf = vim.api.nvim_get_current_buf()
+  local expandtab = vim.bo[curbuf].expandtab
+  local base_indent = expandtab and (' '):rep(vim.fn.indent('.')) or ''
```

Comment:
Why is it empty with `noexpandtab`? And instead of manually constructing the indent wouldn't it be better to use `:left` that supports all kind of indent (e.g. `shiftwidth=4 tabstop=8`, `vartabstop`)?

Reply:
yep left make more sense insetead generate a indent